plugins/lua/sort/places.lua

```diff
@@ -0,0 +1,113 @@
+local _ENV = mkmodule('plugins.sort.places')
+
+local sortoverlay = require('plugins.sort.sortoverlay')
+local widgets = require('gui.widgets')
+local utils = require('utils')
+
+local info = df.global.game.main_interface.info
+local buildings = info.buildings
+
+local zone_names = {
+    [df.civzone_type.MeetingHall] = 'Meeting Area',
+    [df.civzone_type.Bedroom] = 'Bedroom',
+    [df.civzone_type.DiningHall] = 'Dining Hall',
+    [df.civzone_type.Pen] = 'Pen/Pasture',
+    [df.civzone_type.Pond] = 'Pit/Pond',
+    [df.civzone_type.WaterSource] = 'Water Source',
+    [df.civzone_type.Dungeon] = 'Dungeon',
+    [df.civzone_type.FishingArea] = 'Fishing',
+    [df.civzone_type.SandCollection] = 'Sand',
+    [df.civzone_type.Office] = 'Office',
+    [df.civzone_type.Dormitory] = 'Dormitory',
+    [df.civzone_type.Barracks] = 'Barrachs',
+    [df.civzone_type.ArcheryRange] = 'Archery Range',
+    [df.civzone_type.Dump] = 'Garbage Dump',
+    [df.civzone_type.AnimalTraining] = 'Animal Training',
+    [df.civzone_type.Tomb] = 'Tomb',
+    [df.civzone_type.PlantGathering] = 'Gather Fruit',
+    [df.civzone_type.ClayCollection] = 'Clay'
+}
+
+local function get_default_zone_name(zone_type)
+    return zone_names[zone_type] or ''
+end
+
+local function get_zone_search_key(zone)
+    local site = df.global.world.world_data.active_site[0]
+    local result = {}
+
+    -- allow zones to be searchable by their name
+    if #zone.name == 0 then
+        table.insert(result, get_default_zone_name(zone.type))
+    else
+        table.insert(result, zone.name)
+    end
+
+    -- allow zones w/ assignments to be searchable by their assigned unit
+    if zone.assigned_unit ~= nil then
+        table.insert(result, sortoverlay.get_unit_search_key(zone.assigned_unit))
+    end
+
+    -- allow zones to be searchable by type
+    if zone.location_id == -1 then -- zone is NOT a special location and we don't need to do anything special for type searching
+        table.insert(result, df.civzone_type[zone.type]);
+    else -- zone is a special location and we need to get its type from world data
+        local building, success, _ = utils.binsearch(site.buildings, zone.location_id, 'id')
+        if success and building.name then
+            table.insert(result, df.language_name_type[building.name.type])
```

Comment:
I think you're going to need a different way to map the location type to a searchable string. Nobody is going to search for `SymbolFood` when looking for a Tavern, for example.

Reply:
My first thought would be to create table matching arbitrary `df.language_name_type`s to `df.civzone_type` similar to how I created the `zone_type` table earlier.

Is there a more idiomatic or API-using solution you would recommend?